main.go

```diff
@@ -313,6 +370,46 @@ func (m *Main) generateBatches() <-chan []byte {
 	return ch
 }
 
+type Stats struct {
+	Time   time.Time
+	Tags   map[string]string
+	Fields models.Fields
+}
+
+func (m *Main) GatherStats() *Stats {
+	m.mu.Lock()
+	elapsed := time.Since(m.now).Seconds()
+	pThrough := float64(m.writtenN) / elapsed
+	s := &Stats{
+		Time: time.Unix(0, int64(time.Since(m.now))),
+		Tags: m.ReportTags,
+		Fields: models.Fields(map[string]interface{}{
+			"T":              int(elapsed),
+			"points_written": m.writtenN,
+			"values_written": m.writtenN * m.FieldsPerPoint,
+			"points_ps":      pThrough,
+			"values_ps":      pThrough * float64(m.FieldsPerPoint),
+			"write_error":    m.currentErrors,
+			"resp_wma":       int(m.wmaLatency),
+			"resp_mean":      int(m.totalLatency) / len(m.latencyHistory) / int(time.Millisecond),
+			"resp_90":        int(m.quartileResponse(0.9) / time.Millisecond),
+			"resp_95":        int(m.quartileResponse(0.95) / time.Millisecond),
+			"resp_99":        int(m.quartileResponse(0.99) / time.Millisecond),
+		}),
+	}
+
+	var isCreating bool
+	if m.writtenN < m.SeriesN() {
+		isCreating = true
+	}
+	s.Tags["creating_series"] = fmt.Sprint(isCreating)
+
+	// Reset error count for next reporting.
+	m.currentErrors = 0
+	m.mu.Unlock()
+	return s
+}
```

Comment:
Can you just call this `Stats()`?

Reply:
Also, can you defer the `Unlock()`? I don't think the tiny perf boost is worth the potential maintenance issues.